src/test/java/net/imagej/patcher/HeadlessCompletenessTest.java

```diff
@@ -150,7 +150,9 @@ public void missingGenericDialogMethods() throws Exception {
 			"windowDeactivated(java.awt.event.WindowEvent)", //
 			"windowDeiconified(java.awt.event.WindowEvent)", //
 			"windowIconified(java.awt.event.WindowEvent)", //
-			"windowOpened(java.awt.event.WindowEvent)" })
+			"windowOpened(java.awt.event.WindowEvent)",
+			"addButton(java.lang.String,java.awt.event.ActionListener)"
```

Comment:
Annotating this publicly for possible future reference, the reason we can skip over `addButton` is that `addButton` has a `GraphicsEnvironment.isHeadless()` check and returns before it does anything with graphics.
https://github.com/imagej/ImageJ/blame/3135feb74aee01f9acf49b0ed45bf7fcd0114d22/ij/gui/GenericDialog.java#L387-L389
```java
	public void addButton(String label, ActionListener listener) {
		if (GraphicsEnvironment.isHeadless())
			return;
```

Reply:
@mikitti Thank you very much for your analysis. Best would be to add the annotations as comments to the source code of ij1-patcher, so that anyone reading the code later has the explanation right there, rather than needing to dig on GitHub, where it will be harder to find (and may even someday be lost).